lib/heap_sort.rb

```diff
@@ -1,8 +1,34 @@
-
+# require_relative "min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(n log n)
+# Space Complexity: O(n)
 def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+  return [] if list.empty?
+  return list if list.length == 1
+
+  # build heap 
+  heap = MinHeap.new()
+  list.each do |node|
+    heap.add(node)
+  end 
+
+  # tried doing it the way many others did but last test kept failing
+  # so did it this way
+  # then we order heap by comparing nodes
+  sorted_list = []
+  list.size.times do |node|
+    last_node = heap.remove()
+    if sorted_list.empty? 
+      sorted_list.push(last_node)
+    elsif last_node < sorted_list[-1]
+      temp = sorted_list[-1]
+      sorted_list[-1] = last_node
+      sorted_list.push(temp)
+    else 
+      sorted_list.push(last_node)
+    end 
+  end 
```

Comment:
This is a bit overly complicated.  Remember you always remove the smallest element of a heap.

```suggestion
  until heap.empty?
    sorted_list << heap.remove()
  end
```